src/modules/authentication/components/ChangePassword/ChangePassword.tsx

```diff
@@ -1,5 +1,212 @@
+import { useEffect, useState } from "react";
+import logo from "../../../../assets/images/PMS-logo.png";
+import { useForm } from "react-hook-form";
+import { PasswordValidation } from "../../../../validations";
+import { toast } from "react-toastify";
+import { AUTH_URLS, axiosInstance, HEADERS } from "../../../../api";
+
+import { useNavigate } from "react-router-dom";
+
 function ChangePassword() {
-	return <div></div>;
+  interface formDataInputs {
+    oldPassword: string;
+    newPassword: string;
+    confirmNewPassword: string;
+  }
+
+  let {
+    register,
+    formState: { errors, isSubmitting },
+    handleSubmit,
+    watch,
+    trigger,
+  } = useForm<formDataInputs>({ mode: "onChange" });
+
+  let navigate = useNavigate();
+
+  let [showPassword, setShowPassword] = useState<boolean>(false);
+  const togglePasswordVisibility = (): void => {
+    setShowPassword((prevState) => !prevState);
+  };
+
+  const onSubmit = async (data: formDataInputs): Promise<void> => {
+    console.log(data);
+    try {
+      let response = await axiosInstance.put<string>(
+        AUTH_URLS.changePassword,
+        data,
+        HEADERS
+      );
+      toast.success("Password changed successfully");
+      navigate("/login");
+    } catch (error: any) {
+      toast.error(error.response.data.message);
+    }
+  };
+
+  let password = watch("newPassword");
+  let confirmPassword = watch("confirmNewPassword");
+  useEffect(() => {
+    if (confirmPassword) trigger("confirmNewPassword");
+  }, [password, confirmPassword, trigger]);
+
+  return (
+    <>
+      <div className="container-fluid change-pass-bg">
+        <div className="row justify-content-center align-items-center change-pass-container">
+          <div className="col-md-6 text-center">
+            <img
+              src={logo}
+              alt="logo"
+              style={{ width: "312px", height: "91px" }}
+            />
+            <div className="change-pass-form-wrapper">
+              <p className="form-welcome">welcome to PMS</p>
+              <h3 className="form-title">
+                <span style={{ textDecorationLine: "underline" }}>C</span>hange
+                password
+              </h3>
+              <form onSubmit={handleSubmit(onSubmit)} className=" text-start">
+                {/*old password */}
+                <div>
```

Comment:
We can extract the PasswordInput with its logic into a separte component and re-use it when needed in our application.